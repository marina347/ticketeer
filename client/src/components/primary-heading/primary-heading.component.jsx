import React from "react";
import "./primary-heading.styles.scss";

const PrimaryHeading = ({ mainText, subText }) => {
  return (
    <h1 className="primary-heading">
      <span className="primary-heading__main">{mainText}</span>
      <span className="primary-heading__sub">{subText}</span>
    </h1>
  );
};

export default PrimaryHeading;
