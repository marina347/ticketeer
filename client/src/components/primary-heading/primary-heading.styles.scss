.primary-heading {
  text-transform: uppercase;
  margin-bottom: 50px;
  text-align: center;
  color: var(--color-grey-dark-1);
  font-weight: 200;
  &__main {
    display: block;
    font-size: 6.5rem;
    letter-spacing: 2.5rem;
    margin-bottom: 1.5rem;
    animation: moveToRight 1s ease-out;
  }

  &__sub {
    display: block;
    font-size: 2.5rem;
    letter-spacing: 1rem;
    animation: moveToLeft 1s ease-out;
  }
}
