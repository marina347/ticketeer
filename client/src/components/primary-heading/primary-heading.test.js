import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";
import { PrimaryHeading } from "./primary-heading.component";

describe("PrimaryHeading component", () => {
  let wrapper;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      mainText: "Main text",
      subText: "Sub text",
    };
    wrapper = shallow(<PrimaryHeading {...mockProps} />);
  });

  it("Should render PrimaryHeading component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render main heading text", () => {
    const primaryHeadingText = wrapper
      .find('[id="primary_heading_main"]')
      .text();
    expect(primaryHeadingText).toBe("Main text");
  });

  it("Should render main heading text", () => {
    const subHeadingText = wrapper.find('[id="primary_heading_sub"]').text();
    expect(subHeadingText).toBe("Sub text");
  });
});
