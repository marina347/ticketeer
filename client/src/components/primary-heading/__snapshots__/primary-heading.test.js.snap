// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryHeading component Should render PrimaryHeading component 1`] = `
<h1
  className="primary-heading"
>
  <span
    className="primary-heading__main"
    id="primary_heading_main"
  >
    Main text
  </span>
  <span
    className="primary-heading__sub"
    id="primary_heading_sub"
  >
    Sub text
  </span>
</h1>
`;
